utils/firestore.js

```diff
@@ -1410,136 +1410,63 @@ const getSpecimenCollections = async (token, siteCode) => {
-            }
-        }
-        else{
-            if(startDate !== "0"){
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('959708259', '==', trackingId).where('656548982', '>=', startDate).orderBy(orderByField, 'desc').limit(elementsPerPage).offset(currPage*elementsPerPage).get();
-            }
-            else{
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('959708259', '==', trackingId).orderBy(orderByField, 'desc').limit(elementsPerPage).offset(currPage*elementsPerPage).get();
-            }
-        }
+
+    if (startDate !== '0') {
+        query = query.where('656548982', '>=', startDate);
     }
-    else{
-        if(endDate !== "0"){
-            if(startDate !== "0"){
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', "<=", endDate).where('656548982' ,">=", startDate).orderBy(orderByField, 'desc').limit(elementsPerPage).offset(currPage*elementsPerPage).get();
-            }
-            else{
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', "<=", endDate).orderBy(orderByField, 'desc').limit(elementsPerPage).offset(currPage*elementsPerPage).get();
-            }
-        }
-        else{
-            if(startDate !== "0"){
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', ">=", startDate).orderBy(orderByField, 'desc').limit(elementsPerPage).offset(currPage*elementsPerPage).get();
-            }
-            else{
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).orderBy(orderByField, 'desc').limit(elementsPerPage).offset(currPage*elementsPerPage).get();
-            }
-        }
+
+    if (source !== 'bptlShippingReport') {
+        query = query.where('789843387', '==', siteCode);
     }
-    let result = snapshot.docs.map(document => document.data());
+    return query
+}
+
+const getBoxesPagination = async (siteCode, body) => {
+    const currPage = body.pageNumber;
+    const orderByField = body.orderBy;
+    const elementsPerPage = body.elementsPerPage;
+    const filters = body.filters ?? {};
+    const source = body.source ?? '';
+
+    const startDate = filters.startDate ?? '0';
+    const trackingId = filters.trackingId ?? '';
+    const endDate = filters.endDate ?? '0';
+
+    let query = db.collection('boxes').where('145971562', '==', 353358909);
+
+    query = buildQueryWithFilters(query, trackingId, endDate, startDate, source, siteCode)
+
+    query = query.orderBy(orderByField, 'desc').limit(elementsPerPage).offset(currPage * elementsPerPage);
+
+    const snapshot = await query.get();
```

Comment:
Seems like it would be good to handle the snapshot error case with `try/catch` here.